### ISSUE TITLE
LuajitError: bad light userdata pointer on aarch64

### ISSUE THREAD

**Issue by rtsisyk** [tarantool/tarantool#2712]
On any test:
```
2017-08-24 13:20:11.808 [32673] main/106/console/unix/:/root/tarantool utils.c:923 E> LuajitError: bad light userdata pointer
```

**Comment by rtsisyk** [comment]
Please move to 1.8.x

**Comment by xpk** [comment]
Guys, this is a major blocker for using Tarantool on arm64. Any suggestions on how one would fix / work around this?
LuaJIT developers suggest replacing lightuserdata with FFI bindings (see https://github.com/LuaJIT/LuaJIT/pull/230 and https://github.com/torch/torch7/issues/1035), but not being overly familiar with C/Lua interaction and especially Tarantool internals, I'm not sure how tricky this would be and if there is a better way...

**Comment by waruqi** [comment]
Here is a workaround implementation, It works fine.

https://github.com/xmake-io/xmake/blob/96b0eae26058482a83f5bbf57bab15a834339464/core/src/xmake/prefix.h#L45-L82

**Comment by igormunkin** [comment]
It seems Mike's fixed the issue via LuaJIT/LuaJIT@e9af1ab (at least a bunch of 64-bit tickets has been closed the same day). We'll take a look on this changeset in scope of #5483.

**Comment by igormunkin** [comment]
The issue is resolved in scope of tarantool/luajit@2cacfa8. The submodule was updated in tarantool in 2.9.0-255-ge42d116f3 (e42d116).
